storage/disk/partition_test.go

```diff
@@ -18,11 +18,13 @@ func TestPartitionTrie(t *testing.T) {
 		storage.MustParsePath("/foo/bar"),
 		storage.MustParsePath("/foo/baz/qux"),
 		storage.MustParsePath("/corge"),
+		storage.MustParsePath("/tenants/*/bindings"), // wildcard in the middle
```

Comment:
What should happen if the caller does `write(/tenants/foo/bar/baz, 7)`

```
/tenants/foo/bindings/<n> <--- each of these go their own row 
/tenants/foo/bar          <-- goes to it's own row, value: `{"baz": 7}`
```

What should happen if the caller does `write(/tenants/foo, {"bindings": {...}, "bar": 7, "baz": 8})`?

```
/tenants/foo/bindings/<n> <-- these are own their own rows
/tenants/foo/bar <-- this is on it's own row (value=7)
/tenants/foo/baz <-- this is on it's own row (value=8)
```